app/Policies/UserCommentPolicy.php

```diff
@@ -59,7 +59,11 @@ public function create(User $user, ?User $commentable): bool
             || !$user->hasVerifiedEmail()
             || $commentable->isBlocking($user)
             || !$commentable->UserWallActive
-            || ($commentable->only_allows_contact_from_followers && !$commentable->isFollowing($user))
+            || (
+                $commentable->only_allows_contact_from_followers
+                && !$commentable->isFollowing($user)
+                && !$user->is($commentable)
```

Comment:
Consider moving the `user->is` check up to avoid the query embedded in the `isFollowing` call

Reply:
Good point - updated in latest.